apps/ledger-live-mobile/src/newArch/features/WalletSync/screens/DeviceSelection/index.tsx

```diff
@@ -35,109 +34,99 @@ type ChooseDeviceProps = Props & {
   goToFollowInstructions: (device: Device) => void;
 };
 
-const ChooseDevice: React.FC<ChooseDeviceProps> = ({ isFocused, goToFollowInstructions }) => {
-  const request = useMemo(
-    () => ({
-      appName: TRUSTCHAIN_APP_NAME,
-    }),
-    [],
-  );
-  const action = useAppDeviceAction();
-  const [device, setDevice] = useState<Device | null>();
-  const [isHeaderOverridden, setIsHeaderOverridden] = useState<boolean>(false);
-
-  const navigation = useNavigation<NavigationProps["navigation"]>();
-
-  const onSelectDevice = (device?: Device) => {
-    if (device) setDevice(device);
-  };
-
-  const onResult = useCallback(
-    (payload: AppResult) => {
-      setDevice(undefined);
-
-      // Nb Unsetting device here caused the scanning to start again,
-      // scanning causes a disconnect, which throws an error when we try to talk
-      // to the device on the next step.
-      goToFollowInstructions(payload.device);
-    },
-    [goToFollowInstructions],
-  );
-
-  const onModalHide = () => {
-    setDevice(undefined);
-  };
-
-  const onError = (error: Error) => {
-    // Both the old (SelectDevice) and new (SelectDevice2) device selection components handle the bluetooth requirements with a hook + bottom drawer.
-    // By setting back the device to `undefined` it gives back the responsibilities to those select components to check for the bluetooth requirements
-    // and avoids a duplicated error drawers/messages.
-    // The only drawback: the user has to select again their device once the bluetooth requirements are respected.
-    if (error instanceof BluetoothRequired) {
-      setDevice(undefined);
-    }
-  };
-
-  const requestToSetHeaderOptions = useCallback(
-    (request: SetHeaderOptionsRequest) => {
-      if (request.type === "set") {
-        navigation.setOptions({
-          headerShown: true,
-          headerLeft: request.options.headerLeft,
-          headerRight: request.options.headerRight,
-          title: "",
-        });
-        setIsHeaderOverridden(true);
-      } else {
-        // Sets back the header to its initial values set for this screen
-        navigation.setOptions({
-          headerLeft: () => null,
-          headerRight: () => null,
-          ...headerOptions,
-        });
-        setIsHeaderOverridden(false);
+const request = {
+  appName: TRUSTCHAIN_APP_NAME,
+};
+
+const WalletSyncActivationDeviceSelection: React.FC<ChooseDeviceProps> = memo(
```

Comment:
the component is getting `memo` twice, you can probably remove this one and keep the one at the export side.